File: src/main/java/com/sothawo/akkabatch/RecordProcessor.java
package com.sothawo.akkabatch;

import java.util.Random;

/**
 * Datensatzverarbeitung. Liefert eine Kopie des Record und verbraucht etwas Zeit.
 */
public final class RecordProcessor {
// ------------------------------ FIELDS ------------------------------

    private static Random rand = new Random();

// -------------------------- STATIC METHODS --------------------------

    public static Record processRecord(final Record origin) {
        if (null == origin) {
            throw new IllegalArgumentException("null input");
        }
        Record record =
                new Record(origin.getId(), origin.getSex(), origin.getFirstname(), origin.getLastname(),
                           origin.getZip(), origin.getCity(), origin.getDistrict(), origin.getStreet(),
                           origin.getNumber());
        if (0 == (rand.nextInt() % 2)) {
            try {
                Thread.sleep(20);
            } catch (InterruptedException ignored) {
                System.err.println("oops, interrupted");
            }
        } else {
            Fibonacci.calculate(15000);
        }
        return record;
    }
}

File: src/main/scala/com/sothawo/akkabatch/scala/RecordProcessor.scala
package com.sothawo.akkabatch.scala

import scala.beans.BeanProperty
import scala.util.Random

/**
 * Dummy record processing. creates a copy and uses some time.
 *
 * @author P.J. Meisch (pj.meisch@sothawo.com).
 */
object RecordProcessor {
  @BeanProperty var fibonacci: Int = 1000
  @BeanProperty var threadsleep: Int = 10

  def processRecord(origin: Record): Record = {
    if (null == origin) {
      throw new IllegalArgumentException("null input")
    }

    useTime
    origin.copy()
  }

  /**
   * use some time
   */
  def useTime {
    if (0 == (Random.nextInt % 2)) {
      try {
        Thread.sleep(threadsleep)
      }
      catch {
        case ignored: InterruptedException => {
          System.err.println("oops, interrupted")
        }
      }
    }
    else {
      Fibonacci.calculate(fibonacci)
    }
  }
}
